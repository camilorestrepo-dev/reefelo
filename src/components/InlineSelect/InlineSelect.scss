@import "../../styles/shared.scss";

.inline-select-container {
  label {
    display: block;
    margin-bottom: $spacing-4;
  }

  .options-container {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    gap: $spacing-4;

    .custom-btn {
      &:not(.selected) {
        color: $grey;
        border-color: $grey;
      }

      &.selected {
        color: $text-color;
        border-color: $text-color;
        border-width: 2px;
      }
    }
  }
}