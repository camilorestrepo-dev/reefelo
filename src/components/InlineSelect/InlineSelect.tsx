import "./InlineSelect.scss";
import { useState } from "react";
import classnames from "classnames";
import Button from "../Button/Button";

type InlineOption = {
  label: string;
  value: number | string;
};

type InlineSelectProps = {
  className?: string;
  value?: number | string;
  label?: string;
  options: InlineOption[];
  onChange?: (newOptionValue: number | string) => void;
};

const InlineSelect: React.FC<InlineSelectProps> = ({
  className,
  value,
  label,
  options,
  onChange,
}: InlineSelectProps) => {
  const [selectedOption, setSelecterOption] = useState(value);

  const changeOption = (option: InlineOption) => {
    const value = option.value;
    setSelecterOption(value);
    onChange(value);
  };

  return (
    <div className={classnames("inline-select-container", className)}>
      {label && <label>{label}</label>}
      <div className="options-container">
        {options.map((option) => (
          <Button
            variant="outline"
            key={`option-${option.value}`}
            className={classnames({ selected: selectedOption === option.value })}
            onClick={() => changeOption(option)}
          >
            {option.label}
          </Button>
        ))}
      </div>
    </div>
  );
};

export default InlineSelect;
