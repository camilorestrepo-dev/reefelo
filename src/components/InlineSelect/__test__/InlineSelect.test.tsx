import { render, screen, fireEvent } from "@testing-library/react";
import InlineSelect from "../InlineSelect";

test("render correctly", () => {
  const mockFn = () => {};
  const component = render(
    <InlineSelect
      label="Tickets Options"
      defaultValue={10}
      options={[
        { label: "10", value: 10 },
        { label: "100", value: 100 },
        { label: "1000", value: 1000 },
      ]}
      onChange={mockFn}
    />,
  );
  expect(component.container).toMatchSnapshot();
});

test("click in the right option", async () => {
  let pressedOption: number;
  const onPressed = (id: number) => {
    pressedOption = id;
  };
  render(
    <InlineSelect
      label="Tickets Options"
      defaultValue={10}
      options={[
        { label: "10", value: 10 },
        { label: "100", value: 100 },
        { label: "1000", value: 1000 },
        { label: "10000", value: 10000 },
      ]}
      onChange={onPressed}
    />,
  );

  expect(screen.getByText("Tickets Options")).toBeInTheDocument();

  const optionBtn = screen.getByText("10000");
  expect(optionBtn).toBeInTheDocument();

  fireEvent.click(optionBtn);
  expect(pressedOption).toBe(10000);
});
