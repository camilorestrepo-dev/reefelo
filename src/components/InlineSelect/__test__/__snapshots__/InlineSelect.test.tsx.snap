// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div>
  <div
    class="inline-select-container"
  >
    <label>
      Tickets Options
    </label>
    <div
      class="options-container"
    >
      <button
        class="custom-btn outline-btn selected"
        type="button"
      >
        10
      </button>
      <button
        class="custom-btn outline-btn"
        type="button"
      >
        100
      </button>
      <button
        class="custom-btn outline-btn"
        type="button"
      >
        1000
      </button>
    </div>
  </div>
</div>
`;
